### ISSUE TITLE
About Refinement Transformer

### ISSUE THREAD

**Issue by Seoneun** [exitudio/BAMM#3]
Great works!
I hope to see all codes of this project soon.

I have question about Refinement Transformer.

You mentioned that BAMM used RVQ architecture and also used Refinement Transformer for generating another tokens of base tokens.

Then, I want to know how train it.

Similar to MoMask, did you train Refinement Transformer freezing Masked Self-attention Transformer after training Masked Self-attention Transformer?

Or did you train Refinement Transformer and Masked Self-attention Transformer at the same time?

Thank you!

**Comment by exitudio** [comment]
Hi, we use pretrain residual layers from MoMask.

**Comment by Seoneun** [comment]
You're saying that RVQ was taken from MoMask, right?

I have a question. For generating tokens for the residual layers, the Residual Transformer (R-Transformer) is used in MoMask. In BAMM paper, it says "The refinement transformer shares the same architecture as the masked self-attention transformer with a full attention mask." Does this mean that the weights are shared and they are trained simultaneously, or is it that the base tokens are first learned with the Masked Self-Attention Transformer, and then the residual tokens are learned with the Refinement Transformer?

Thank you.

**Comment by exitudio** [comment]
- Yes. We use pretrained RVQ from MoMask.
- This is correct "_base tokens are first learned with the Masked Self-Attention Transformer, and then the residual tokens are learned with the Refinement Transformer_"
- _"The refinement transformer shares the same ... "_, The architecture (num head, embedding size, etc) is the same. But don't share weights and RVQ has no mask attention (neither Bidirectional Causal Mask nor Unidirectional Causal Mask) since all base tokens are already generated.
